File: feinblelib/src/main/java/com/bbg/feinblelib/callback/BleReadCallback.java
package com.bbg.feinblelib.callback;

import com.bbg.feinblelib.exception.BleException;

import java.util.HashMap;

public abstract class BleReadCallback extends BleBaseCallback {

    public abstract void onReadSuccess(HashMap data);

    public abstract void onReadFailure(BleException exception);

}

File: feinblelib/src/main/java/com/bbg/feinblelib/callback/BleReadCallback.kt
package com.bbg.feinblelib.callback

import com.bbg.feinblelib.exception.BleException
import java.util.*

abstract class BleReadCallback : BleBaseCallback() {
    abstract fun onReadSuccess(data: HashMap<*, *>?)
    abstract fun onReadFailure(exception: BleException?)
}